src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -3597,7 +3597,7 @@ QString QgsProcessingParameterMatrix::valueAsPythonString( const QVariant &value
         else if ( v2.toString().isEmpty() )
           parts2 << QStringLiteral( "''" );
         else
-          parts2 << v2.toString();
+          parts2 << QgsProcessingUtils::stringToPythonLiteral( v2.toString() );
```

Comment:
I think this should be variantToPythonLiteral instead, to avoid the forced string conversions